src/cmap/connection_pool.ts

```diff
@@ -735,7 +746,7 @@ export class ConnectionPool extends TypedEventEmitter<ConnectionPoolEvents> {
     }
   }
 
-  private processWaitQueue() {
+  private processWaitQueue(options?: { timeout?: Timeout }) {
```

Comment:
NOTE: this is added to allow us to pass timeouts through to `ConnectionPool.createConnection` in order to satisfy the CSOT spec's requirement to apply the serverSelectionTimeout to the creation of new connections when necessary